var/spack/repos/builtin/packages/unzip/package.py

```diff
@@ -0,0 +1,46 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Unzip(MakefilePackage):
+    """Unzip is a compression and file packaging/archive utility."""
+
+    homepage = 'http://www.info-zip.org/Zip.html'
+    url      = 'http://downloads.sourceforge.net/infozip/unzip60.tar.gz'
+
+    version('6.0', '62b490407489521db863b523a7f86375')
+
+    conflicts('platform=cray', msg='Unzip does not currently build on Cray')
+
+    make_args = ['-f', 'unix/Makefile']
```

Comment:
If you replace this with `build_directory = 'unix'` you won't need to use it anywhere else. This is how `MakefilePackage` handles Makefiles that live in a subdirectory.

Reply:
I get this if I do so:
```
==> Error: ProcessError: Command exited with status 2:
    'make' '-j8' 'generic'

1 error found in build log:
     4    ==> 'make' '-j8' 'generic'
  >> 5    make: *** No rule to make target `unix/configure', needed by `flags'.  Stop.
```

Reply:
Interesting, it looks like you have to be in the main directory to run the Makefile at all. What a poorly designed Makefile...